server/lib/logger.ts

```diff
@@ -171,3 +225,30 @@ export function log(location, message, level) {
       )
     );
 }
+
+export async function addFrontendLog(location, message, level) {
+  return await initDirectory()
+    .then(() => {
+      if (allowed) {
+        checkFrontendLogFiles();
+        wazuhFrontendLogger.log({
+          date: new Date(),
+          level: level || 'error',
+          location: location || 'Unknown origin',
+          message: message || 'An error occurred'
+        });
```

Comment:
put this structure into a type or interface, then use this in the args of the functions and his consumers